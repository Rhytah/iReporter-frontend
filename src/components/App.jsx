import React from 'react';
import { Provider } from 'react-redux';
import { toast, ToastContainer } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';
import store from '../store';
import Routes from '../Routes';

// we will probably need to notify in very many places , so it makes sense to have it available to us globaly
window.Notify = toast;

export default function App() {
  return (
    <Provider store={store}>
      <React.Fragment>
        <ToastContainer />
        <Routes />
      </React.Fragment>
    </Provider>
  );
}
