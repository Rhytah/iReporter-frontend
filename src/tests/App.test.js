import React from 'react';
import { shallow } from 'enzyme';
import App from '../components/App';

describe('App Component', () => {
  it('matches the snapshot', () => {
    const component = shallow(<App />);
    expect(component).toMatchSnapshot();
  });
});
