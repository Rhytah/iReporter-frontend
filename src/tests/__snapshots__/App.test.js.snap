// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component matches the snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <div>
    <ToastContainer
      autoClose={5000}
      bodyClassName={null}
      className={null}
      closeButton={
        <CloseButton
          ariaLabel="close"
        />
      }
      closeOnClick={true}
      containerId="A"
      draggable={true}
      draggablePercent={80}
      enableMultiContainer={true}
      hideProgressBar={false}
      newestOnTop={false}
      pauseOnFocusLoss={true}
      pauseOnHover={true}
      position="top-right"
      progressClassName={null}
      progressStyle={null}
      role="alert"
      rtl={false}
      style={null}
      toastClassName={null}
      transition={[Function]}
    />
    <Routes />
  </div>
</Provider>
`;
